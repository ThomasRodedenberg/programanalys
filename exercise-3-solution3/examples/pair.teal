type Pair(fst : nonnull int, snd : int);

fun null_producer(n : int) = {
    if (n == 0) {
       return null;
    }
    return n;
}

fun main(n : int) : nonnull int = {
    var p : Pair := new Pair(0, 1);
    p.snd := null;
    p.fst := null_producer(n);
    return p.fst;
}
