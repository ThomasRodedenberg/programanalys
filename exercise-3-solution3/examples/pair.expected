merge: 1 10 29
N 1 10 -> null 5 14
N 1 10 -> obj 11 29
N 1 10 -> obj 11 32
N 1 10 -> null 12 13
N 14 11 -> null 5 14
N 14 11 -> obj 11 29
N 14 11 -> obj 11 32
N 14 11 -> null 12 13
